File: 백준/Bronze/15829. Hashing/Hashing.cc
#include<iostream>
#include <cmath>

using namespace std;
int main(){
    int len;
    cin>>len;
    long long sum=0;
    long long r=1;
    long long  M = 1234567891;
    string num;
    cin>>num;
    for(int i=0;i<num.size();i++){
        sum= (sum+ (num[i]-96)* r)%M;
        r = (r* 31) % M;
    }
    cout<<sum<<"\n";
}

File: 백준/Bronze/15829. Hashing/Hashing.py

if __name__ == '__main__':
    # print(ord('a')-96)
    L = int(input())
    num = input()
    sum = 0
    for i in range(0,len(num)):
        sum += pow(31,i) * (ord(num[i]) - 96)
    print(sum %1234567891)




